### PR TITLE
Use infinity-agent if defaultMaxSockets is 5

### PR DESCRIPTION
**Pull request by floatdrop**
If user wants to limit open sockets - got should not interfere.

### CODE REVIEW

**Comment by julien-f** [comment]
Too much magic for me…

Users should set the agent if they need to limit the number of connections.

**Comment by sindresorhus** [comment]
@julien-f Just badly worded. Check the PR for better description: https://github.com/floatdrop/infinity-agent/pull/1

File: readme.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ Milliseconds after which the request will be aborted and an error event with `ET
 
 [http.Agent](http://nodejs.org/api/http.html#http_class_http_agent) instance.
 
-Node HTTP/HTTPS [Agent in 0.10.35](https://github.com/joyent/node/blob/v0.10.35-release/lib/http.js#L1261) by default limits number of open sockets to 5 — which is a too low. By default `got` will use `new Agent({maxSockets: Infinity})` like new [`Agent` in 0.11.14](https://github.com/joyent/node/blob/v0.11.14-release/lib/_http_agent.js#L110).
+Node HTTP/HTTPS [Agent in 0.10.35](https://github.com/joyent/node/blob/v0.10.35-release/lib/http.js#L1261) by default limits number of open sockets to 5 — which is a too low. If `http.Agent.defaultMaxSockets` is 5 - `got` will use [infinity-agent](https://github.com/floatdrop/infinity-agent), like new [`Agent` in 0.11.14](https://github.com/joyent/node/blob/v0.11.14-release/lib/_http_agent.js#L110).
```
Comment (sindresorhus) [inline readme.md]: This doesn't sound right.

**Comment by julien-f** [comment]
@sindresorhus I get it now, seems a good idea.

File: readme.md (hunk 1 above)
Comment (floatdrop) [inline readme.md]: Try'd to reword this, but still sounds weird.

**Pull request merged**